File: types/season-summary.ts
export type PrizeType = "first_place" | "second_place" | "longest_survivor" | "highest_total_points"

export type PrizeWinner = {
  prize: PrizeType
  prizeName: string
  icon: string
  userId: string
  playerName: string
  stat: string
  payout?: string
}

export type FinalStanding = {
  rank: number
  userId: string
  playerName: string
  pointsAtElimination: number
  totalPoints: number
  strikes: number
  weekEliminated: number | null
}

export type SeasonSummary = {
  isLeagueEnded: boolean
  prizes: PrizeWinner[]
  standings: FinalStanding[]
}

File: api/app/models/season_summary.py
"""Port of types/season-summary.ts. Confirmed-accurate per CR-105-FINDINGS.md
Table 4 -- all four PrizeType literals and both PrizeWinner/FinalStanding shapes
match lib/db.ts:1830-1885 (getSeasonSummary) field-for-field."""
from typing import List, Literal, Optional

from pydantic import BaseModel

PrizeType = Literal["first_place", "second_place", "longest_survivor", "highest_total_points"]


class PrizeWinner(BaseModel):
    prize: PrizeType
    prizeName: str
    icon: str
    userId: str
    playerName: str
    stat: str
    payout: Optional[str] = None


class FinalStanding(BaseModel):
    rank: int
    userId: str
    playerName: str
    pointsAtElimination: int
    totalPoints: int
    strikes: int
    weekEliminated: Optional[int] = None


class SeasonSummary(BaseModel):
    isLeagueEnded: bool
    prizes: List[PrizeWinner]
    standings: List[FinalStanding]
